packages/legacy-workbench/src/js/angular/ttyg/directives/agent-select-menu.directive.js

```diff
@@ -83,8 +84,33 @@ function AgentSelectMenuComponent(TTYGContextService, $translate, $sce, ModalSer
              */
             const onAgentListChanged = (agents) => {
                 buildAgentOptionsModel(agents);
+                selectLastUsedOrDefaultAgent();
             };
 
+            /**
+             * Function to select the most recently used agent (according to local store) or selects the first in the list
+             * if no recent agent is available.
+             *
+             * @function
+             */
+            const selectLastUsedOrDefaultAgent = () => {
+                let selectedAgent;
+
+                // If saved in the LocalStore, the user has selected it, so select it here
+                $scope.agentOptionsList.forEach((agent) => {
+                    if (agent.data.agent.id === TTYGStorageService.getAgentId()) {
```

Comment:
Get the agent id once, outside of the for loop

Reply:
Took it out and added a check if defined.